rai/node/lmdb.cpp

```diff
@@ -1829,12 +1829,12 @@ void rai::mdb_store::checksum_del (rai::transaction const & transaction_a, uint6
 
 void rai::mdb_store::flush (rai::transaction const & transaction_a)
 {
-	std::unordered_map<rai::account, std::shared_ptr<rai::vote>> sequence_cache_l;
 	{
 		std::lock_guard<std::mutex> lock (cache_mutex);
-		sequence_cache_l.swap (vote_cache);
+		vote_cache_l2 = vote_cache_l1;
+		vote_cache_l1.clear ();
 	}
-	for (auto i (sequence_cache_l.begin ()), n (sequence_cache_l.end ()); i != n; ++i)
+	for (auto i (vote_cache_l2.begin ()), n (vote_cache_l2.end ()); i != n; ++i)
```

Comment:
Actually accessing l2 here isn't safe because cache_mutex is no longer locked.

Reply:
It's indeed safe because the function won't be called concurrently, and the function is the only place where it's written to.

Reply:
Good point, looks right.

Reply:
Good point, looks right.